contents/data_vis_aog.md

```diff
@@ -0,0 +1,234 @@
+# Data Visualization with AlgebraOfGraphics.jl {#sec:aog}
+
+`AlgebraOfGraphics.jl` is a higher-level plotting package that uses `Makie.jl` under the hood.
+It is geared towards data visualization workflows with support for `DataFrame`s.
+`AlgebraOfGraphics.jl` abstracts away some common patterns in data visualization using an interface inspired by R's `ggplot2` package.
+
+`AlgebraOfGraphics.jl` follows a layered approach to construct visualizations in a structured manner.
+There are four main types of layers:
+
+- **data** layer
+- **mapping** layer
+- **visual transformation** layer
+- **statistical transformation** layer
+
+> **_NOTE:_**
+> `AlgebraOfGraphics.jl` has some guiding principles described in their [philosophy section](https://aog.makie.org/stable/philosophy/).
+
+`AlgebraOfGraphics.jl` allows you to construct all of these types of layers with functions that returns a `Layer` object,
+in which all of the information necessary will be encoded.
+You can then perform **two operations on layers**:
+
+- **multiplication with `*`**: this fuses two or more layers into a _single_ layer
+- **addition with `+`**: this superimposes two or more layers into a vector of `Layer`s
+
+Finally, as the name suggests, `AlgebraOfGraphics.jl` is an algebra for `Layer` objects.
+And, as such, it defines two algebraic properties.
+Let `a`, `b` and `c` be `Layer` objects, then the following two properties hold:
+
+- **associative property**: `(a * b) * c = a * (b * c)`
+- **distributive property**: `a  * (b + c) = (a * b) + (a + b)`
+
+To get started with `AlgebraOfGraphics.jl`,
+you'll need to load it along with a desired `Makie.jl` backend (Chapter -@sec:DataVisualizationMakie):
+
+```julia
+using AlgebraOfGraphics
+using CairoMakie
+```
+
+## Layers {#sec:aog_layers}
+
+We'll cover the **data layer** first.
+You can create data layers with the `data` function:
```

Comment:
```suggestion
Which can be created with the `data` function:
```

Reply:
Can we start sentences with `Which`?

Reply:
maybe not the best source. https://www.quickanddirtytips.com/articles/can-you-start-a-sentence-with-which/, but your sentence is also ok. My issue with it is that `data layers` is twice too close.